packages/strapi-connector-bookshelf/lib/buildQuery.js

```diff
@@ -16,7 +16,9 @@ const BOOLEAN_OPERATORS = ['or'];
 const buildQuery = ({ model, filters }) => qb => {
   const joinsTree = buildJoinsAndFilter(qb, model, filters);
 
-  if (_.has(filters, 'where') && Array.isArray(filters.where) && filters.where.length > 0) {
+
+  if (_.has(filters, 'where') && Array.isArray(filters.where) && filters.where.length > 0 &&
+      !(filters.limit && filters.limit == 1) && joinsTree.joins && Object.keys(joinsTree.joins).length) {
```

Comment:
can you put the new condition part into a function to avoid too long if statement